### DISCUSSION TITLE
Possible to reconstruct zeek logs from elasticsearch?

### DISCUSSION THREAD

**Discussion by DebianGuru** [Security-Onion-Solutions/securityonion#7630] [Unsupported Versions] [answered]
I have a standalone install on a small network.  It was originally installed where the /nsm folder was stored in one large logical group within LVM.  I generated a lv for the nsm folder by shrinking down the lv first, making a newer, larger lv for the nsm folder and then I mounted it, and moved data from the old /nsm folder in the original lv to where I mounted my new lv.  This worked fine, except I found out that the last few weeks of my /nsm/zeek/logs folder got deleted.  I'm unsure how this happened.  I think that shrinking the original lv triggered some housekeeping, due to having very little free space.  This is just my theory and I don't have any other ideas how it happened.

At any rate, I used to keep the zeek log files to show my manager that we are indeed keeping some readable "logs" of traffic.  I'm required to keep a year's worth.  Now, I realize that Elasticsearch still has all the data it ingested from that time period.  I'm wondering if it's possible to do the reverse of the normal operation and have Elasticsearch spit back out the info that was ingested into a log.  Essentially, recreating the missing zeek logs.  If this is not possible, what's the next best idea?  Pulling CSVs out from Kibana came to mind, but they're extremely size limited on how much it will let me pull out. 

Thanks in advance for any ideas.

**Comment by dougburks** [comment, marked as answer]
I haven't tried it, but you might be able to use something like elasticsearch-dump:
https://github.com/elasticsearch-dump/elasticsearch-dump
